### PR TITLE
[Debugger] Improve remote debugging support for executable JAR

### LINKED ISSUE #31430: bal run --debug=<port> fails when an executable jar is provided
**Description:**
For any command, the options can be given as `--<option-name> <value>` or `--<option-name>=<value>`. Both formats are supported by default by the Picocli library which is used for the command implementation. 

`bal run` in remote debug mode for Standalone file and package execution can be executed with both formats but since the `bal run <executable-jar>` is handled by the `<bal-dist>/bin/bal` script, AFAIU, when implementing the remote debug with jar execution we have only considered the first format. In order to be consistent, we need to support `bal run --debug=<port> <executable-jar>` as well.

**Steps to reproduce:**
```
bal run --debug 5005 target/bin/helloworld.jar 
Listening for transport dt_socket at address: 5005
```

```
bal run --debug=5005 target/bin/helloworld.jar 
ballerina: Invalid Ballerina source file(.bal): target/bin/helloworld.jar

USAGE:
    bal run [--experimental] [--offline]
                  [<executable-jar | ballerina-file | package-path>] [-- program-args...]

```

**Affected Versions:**
SL Beta.1

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR fixes #31430.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java [hunk 1]
```diff
@@ -86,6 +86,23 @@ public void testSuspendOnBallerinaJarRun() throws BallerinaTestException {
...
     }
 
+    @Test(description = "Tests executable JAR debugging support with [--debug=<PORT>] option.")
+    public void testSuspendOnBallerinaJarRun2() throws BallerinaTestException {
+        String executablePath = Paths.get("target", "bin", testProjectName.replaceAll("-", "_") + ".jar")
+                .toFile().getPath();
+        LogLeecher clientLeecher = new LogLeecher(executablePath);
+        balClient.runMain("build", new String[]{}, null, new String[]{},
+                new LogLeecher[]{clientLeecher}, debugTestRunner.testProjectPath);
+        clientLeecher.waitForText(20000);
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java:96]: This code segment is repeated in `testSuspendOnBallerinaJarRun1` and `testSuspendOnBallerinaJarRun2`. Shall we move this to a `utils` function?

Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java]: Added with ec57c71

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java [hunk 2]
```diff
@@ -64,25 +67,38 @@ public void testSuspendOnBallerinaFileRun() throws BallerinaTestException {
         String msg = REMOTE_DEBUG_LISTENING + port;
         LogLeecher clientLeecher = new LogLeecher(msg);
         balClient.debugMain("run", new String[]{"--debug", String.valueOf(port),
-            debugTestRunner.testEntryFilePath}, null, new String[]{}, new LogLeecher[]{clientLeecher},
-            debugTestRunner.testProjectPath, 10);
+                        debugTestRunner.testEntryFilePath}, null, new String[]{}, new LogLeecher[]{clientLeecher},
+                debugTestRunner.testProjectPath, 10);
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java:71]: Formatting issue

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java [hunk 3]
```diff
@@ -64,25 +67,38 @@ public void testSuspendOnBallerinaFileRun() throws BallerinaTestException {
...
+    @Test(description = "Tests executable JAR debugging support with [--debug=<PORT>] option.")
+    public void testSuspendOnBallerinaJarRun2() throws BallerinaTestException {
+        int port = findFreePort();
+        testBalJarInDebugMode("--debug=" + port);
+    }
+
+    public void testBalJarInDebugMode(String... debugOptions) throws BallerinaTestException {
         String executablePath = Paths.get("target", "bin", testProjectName.replaceAll("-", "_") + ".jar")
-            .toFile().getPath();
+                .toFile().getPath();
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java:89]: Formatting issue

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java [hunk 4]
```diff
@@ -64,25 +67,38 @@ public void testSuspendOnBallerinaFileRun() throws BallerinaTestException {
...
         LogLeecher clientLeecher = new LogLeecher(executablePath);
         balClient.runMain("build", new String[]{}, null, new String[]{},
-            new LogLeecher[]{clientLeecher}, debugTestRunner.testProjectPath);
+                new LogLeecher[]{clientLeecher}, debugTestRunner.testProjectPath);
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java:92]: Formatting issue

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java (hunk 3 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java:89]: I just corrected the formatting using wso2 styling guidelines.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java (hunk 4 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/remote/BallerinaRunRemoteDebugTest.java:92]: I just corrected the formatting using wso2 styling guidelines.

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**